### PR TITLE
 Fix the power and space status condition

### LINKED ISSUE #113: spaceStatus and pwrStatus ERROR not working
https://github.com/kc9wwh/macOSUpgrade/blob/326a5bce0a530746ad77b1443ff05f08ed0b07e7/macOSUpgrade.sh#L376

In my testing, Macs that have a spaceStatus=ERROR are not getting the "Requirements Not Met" jamfHelper dialog. Instead, the logs show the "Disk Check: ERROR" and free space amount before proceeding to attempt the upgrade. I do not believe that pwrStatus=ERROR was causing the "Requirements Not Met" dialog either. 

I changed line 376 to the following and believe it's working as expected:
`if ! { [ ${pwrStatus} == "OK" ] && [ ${spaceStatus} == "OK" ]; }; then`

I'm completely new to Github so I apologize if I'm reporting this incorrectly.

### PR DESCRIPTION
**Pull request by kenchan0130**
## Done

- Refactor conditions
  - To make the condition statement syntax uniform
- Fix the power and space status condition
  - This change is main

Fix #113

### CODE REVIEW

File: macOSUpgrade.sh [hunk 1]
```diff
@@ -373,7 +373,7 @@ fi
 # APPLICATION
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
-if ! [[ ${pwrStatus} == "OK" ]] && [[ ${spaceStatus} == "OK" ]]; then
+if [ "$pwrStatus" != "OK" ] || [ "$spaceStatus" != "OK" ]; then
```
Comment (kenchan0130) [inline macOSUpgrade.sh]: There was a problem in applying the negative statement.

**Comment by kenchan0130** [comment]
@taniguti Please check and review this PR.

File: macOSUpgrade.sh [hunk 2]
```diff
@@ -107,7 +107,7 @@ fi
 ##Full Screen by default
 ##Use Parameter 9 in the JSS.
 userDialog="$9"
-if [[ ${userDialog:=0} != 1 ]]; then userDialog=0 ; fi
+if [[ ${userDialog:=0} -ne 1 ]]; then userDialog=0 ; fi
```
Comment (taniguti) [inline macOSUpgrade.sh]: You would like here should be `[  ${userDialog:=0} -ne 1 ]`

File: macOSUpgrade.sh [hunk 3]
```diff
@@ -212,7 +212,7 @@ fvStatus=$( /usr/bin/fdesetup status | head -1 )
 
 ##Check if device is on battery or ac power
 pwrAdapter=$( /usr/bin/pmset -g ps )
-if [[ ${pwrAdapter} == *"AC Power"* ]]; then
+if [[ "$pwrAdapter" == *"AC Power"* ]]; then
     pwrStatus="OK"
```
Comment (taniguti) [inline macOSUpgrade.sh]:
```
$ /usr/bin/pmset -g ps
Now drawing from 'AC Power'
 -InternalBattery-0 (id=3866723)	100%; charged; 0:00 remaining present: true
$
```

How about like this?
```
$/usr/bin/pmset -g ps | head -1 | tr -d \' | awk '{print $(NF -1)}'
AC
$
```

```
pwrAdapter=$( /usr/bin/pmset -g ps  | /usr/bin/head -1 | /usr/bin/tr -d \' | /usr/bin/awk '{print $(NF -1)}' )
if [  "$pwrAdapter" = AC ]; then
   pwrStatus="OK"
```

File: macOSUpgrade.sh (hunk 1 above)
Comment (taniguti) [reply macOSUpgrade.sh]: LGTM

File: macOSUpgrade.sh (hunk 3 above)
Comment (kenchan0130) [reply macOSUpgrade.sh]: I changed this. https://github.com/kc9wwh/macOSUpgrade/pull/114/commits/ee4d58cdf8e7b100c272db4a869c5b558bbf4cec

File: macOSUpgrade.sh (hunk 2 above)
Comment (kenchan0130) [reply macOSUpgrade.sh]: I changed this. https://github.com/kc9wwh/macOSUpgrade/pull/114/commits/7cb102b03bbd9b411a269224a41f2427ebcdf92f

**Pull request merged**